### PR TITLE
Add describeCondition and entity_ids as additional information to automation step details

### PR DESCRIPTION
**Pull request by LiBe-net**
## Proposed change

Adds some extra information to Automation Traces, such as describeCondition and entity_id 
see https://github.com/home-assistant/frontend/discussions/20660

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

## Summary by CodeRabbit

- **New Features**
	- Enhanced trace path details display with improved context on conditions and entities.
	- Conditional rendering of `alias` and formatted display of `entity_id` for better clarity.

- **Bug Fixes**
	- Improved navigation through nested properties for accurate detail retrieval.

### CODE REVIEW

**Comment by jeeftor** [comment]
Please merge this if it does what I think it might do!

File: src/components/trace/ha-trace-path-details.ts [hunk 1]
```diff
@@ -127,6 +128,19 @@ export class HaTracePathDetails extends LitElement {
...
+      const nestPath = curPath
+        .substring(this.selected.path.length + 1)
+        .split("/");
+      let currentDetail = this.selected.config;
+      for (let i = 0; i < nestPath.length; i++) {
+        if (currentDetail[nestPath[i]] !== undefined) {
+          if (typeof currentDetail[nestPath[i]] !== "string") {
+            currentDetail = currentDetail[nestPath[i]];
+          }
+        }
```
Comment (MindFreeze) [inline src/components/trace/ha-trace-path-details.ts:141]:
```suggestion
        if (!['undefined', 'string'].includes(currentDetail[nestPath[i]])) {
          currentDetail = currentDetail[nestPath[i]];
        }
```

File: src/components/trace/ha-trace-path-details.ts [hunk 2]
```diff
@@ -140,7 +154,9 @@ export class HaTracePathDetails extends LitElement {
 
           return html`
             ${curPath === this.selected.path
-              ? ""
+              ? currentDetail.alias !== undefined
+                ? html`<h2>${currentDetail.alias}</h2>`
+                : ""
```
Comment (MindFreeze) [inline src/components/trace/ha-trace-path-details.ts:159]:
```suggestion
              ? currentDetail.alias
                ? html`<h2>${currentDetail.alias}</h2>`
                : nothing
```

File: src/components/trace/ha-trace-path-details.ts [hunk 3]
```diff
@@ -182,6 +207,12 @@ export class HaTracePathDetails extends LitElement {
             ${Object.keys(rest).length === 0
               ? nothing
               : html`<pre>${dump(rest)}</pre>`}
+            ${currentDetail.entity_id !== undefined &&
+            curPath
+              .substring(this.selected.path.length + 1)
+              .includes("entity_id")
+              ? html`<pre>entity: ${currentDetail.entity_id}</pre>`
+              : ""}
```
Comment (MindFreeze) [inline src/components/trace/ha-trace-path-details.ts:215]:
```suggestion
            ${currentDetail.entity_id &&
            curPath
              .substring(this.selected.path.length + 1)
              .includes("entity_id")
              ? html`<pre>entity: ${currentDetail.entity_id}</pre>`
              : nothing}
```

**Review by MindFreeze — CHANGES_REQUESTED** [review]
A screenshot of before & after is not required but would be very appreciated and it speeds up the review.

File: src/components/trace/ha-trace-path-details.ts [hunk 4]
```diff
@@ -146,6 +160,15 @@ export class HaTracePathDetails extends LitElement {
...
                 </h3>`}
+            ${curPath
+              .substring(this.selected.path.length + 1)
+              .includes("condition")
+              ? html`[${describeCondition(
+                    currentDetail,
+                    this.hass,
+                    currentDetail.alias
+                  )}]<br />`
+              : ""}
```
Comment (MindFreeze) [inline src/components/trace/ha-trace-path-details.ts:171]:
```suggestion
              : nothing}
```

**Review by MindFreeze — CHANGES_REQUESTED** [review]
Once you fix the prettier errors, we can merge this. Thanks!

**Comment by boern99** [comment]
> Once you fix the prettier errors, we can merge this. Thanks!

i fixed the prettier errors ..

**Review by MindFreeze — APPROVED** [review]

**Pull request merged**